sway/desktop/xdg_shell.c

```diff
@@ -358,7 +358,7 @@ static void handle_request_fullscreen(struct wl_listener *listener, void *data)
 	if (e->fullscreen && e->output && e->output->data) {
 		struct sway_output *output = e->output->data;
 		struct sway_workspace *ws = output_get_active_workspace(output);
-		if (ws && !container_is_scratchpad_hidden(container)) {
+		if (ws && container->pending.workspace != ws) {
```

Comment:
`container_is_scratchpad_hidden` also contains a check for `con->scratchpad` -- should we be keeping it?

Reply:
`con->scratchpad` is true for containers that are shown from the scratchpad, so they could belong to a different non-scratchpad workspace. We could just retain the scratchpad hidden check though, like `!container_is_scratchpad_hidden(container) &&` which seems fine.

Leaving it out makes sway honor the output hint for scratchpad hidden containers. I don't think I care either way, but it wasn't really my intention to change the behavior in order to allow that, so I'll add it back in. Good catch.